src/utils/configfile.cc

```diff
@@ -0,0 +1,95 @@
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+
+  Additional permission under GNU GPL version 3 section 7
+
+  If you modify this Program, or any covered work, by linking or
+  combining it with NVIDIA Corporation's libraries from the NVIDIA CUDA
+  Toolkit and the the NVIDIA CUDA Deep Neural Network library (or a
+  modified version of those libraries), containing parts covered by the
+  terms of the respective license agreement, the licensors of this
+  Program grant you additional permission to convey the resulting work.
+*/
+
+#include <fstream>
+#include <iostream>
+
+#include "utils/commandline.h"
+#include "utils/configfile.h"
+#include "utils/optionsparser.h"
+#include "utils/string.h"
+
+namespace lczero {
+  namespace {
+    const char* kConfigFileStr = "Configuration file path";
+    const char* kDefaultConfigFile = ".config";
+  }
+
+std::vector<std::string> ConfigFile::arguments_;
+
+void ConfigFile::PopulateOptions(OptionsParser* options) {
+  options->Add<StringOption>(kConfigFileStr, "config", 'c') = kDefaultConfigFile;
+}
+
+bool ConfigFile::Init(OptionsParser* options) {
+  arguments_.clear();
+
+  // Process flags to get the config file parameter.
+  if (!options->ProcessAllFlags()) return false;
+  
+  // Get the config file parameter passed on the command line or the default.
+  OptionsDict dict = options->GetOptionsDict();
+  std::string filename = dict.Get<std::string>(kConfigFileStr);
+  bool using_default = (filename == kDefaultConfigFile);
+  filename = CommandLine::BinaryDirectory() + "/" + filename;
+
+  // Read the file into the args string.
+  std::string args;
+  if (!ParseFile(filename, args, using_default)) return false;
+
+  arguments_ = StrSplitAtWhitespace(args);
+
+  return true;
+}
+
+bool ConfigFile::ParseFile(std::string filename, std::string& args, bool def) {
+  std::ifstream input(filename);
+
+  if (!input.is_open()) {
+    // It is okay if we cannot open the default file since it is normal
+    // for it to not exist.
+    if (def) return true;
+
+    std::cerr << "Could not open configuration file: " << filename << std::endl;
+    return false;
+  }
+
+  std::cerr << "Found configuration file: " << filename << std::endl;
+
+  for(std::string line; getline( input, line );) {
+    // Remove all leading and trailing whitespace.
+    line = Trim(line);
+    // Ignore comments.
+    if (line.substr(0, 1) == "#") continue;
+
+    args += line + " ";
```

Comment:
Why not append directly to `vector<string>` but instead build a string and then split it?